internal/dataplane/kongstate/consumer_test.go

```diff
@@ -414,41 +424,41 @@ func TestConsumer_SetCredential(t *testing.T) {
 			name: "acl without group",
 			args: args{
 				credType:   "acl",
-				consumer:   &Consumer{Consumer: kong.Consumer{Username: &username}},
+				consumer:   &Consumer{Consumer: kong.Consumer{Username: &username, Tags: []*string{}}},
 				credConfig: map[string]string{},
 			},
-			result:  &Consumer{Consumer: kong.Consumer{Username: &username}},
+			result:  &Consumer{Consumer: kong.Consumer{Username: &username, Tags: []*string{}}},
 			wantErr: true,
 		},
 		{
 			name: "acl with invalid group type",
 			args: args{
 				credType:   "acl",
-				consumer:   &Consumer{Consumer: kong.Consumer{Username: &username}},
+				consumer:   &Consumer{Consumer: kong.Consumer{Username: &username, Tags: []*string{}}},
```

Comment:
There's a lot of changes like this in this PR. What is the specific reason we need this? Maybe we can "guard" ourselves against usage of `nil` (if that's what this is about) and e.g. check for `nil` in `SetCredential()`

Reply:
> What is the specific reason we need this?

A lot of the unit tests are comparing against a complete struct, and the translation functions now populate tags where they weren't previously, which will make `testify.Equal()` reject them. We probably should avoid complete object checks, but refactoring all the tests to check specific fields is a significant amount of work, since we'd need to pick which fields are actually important rather than simply present (a lot of those objects contain default fields we don't really care about). Adding the tags in the complete objects is annoying, but simpler.

A lot of the tags do end up empty because the units don't include the original Kubernetes resource,